applications/common/backend/src/test/java/pl/ds/kyanite/common/components/models/table/TableComponentTest.java

```diff
@@ -37,25 +44,79 @@ public class TableComponentTest {
   @BeforeEach
   public void init() {
     context.addModelsForClasses(TableComponent.class);
-    context.load().json(requireNonNull(
-        Thread.currentThread().getContextClassLoader().getResourceAsStream("table.json")), PATH);
   }
 
   @Test
   void defaultTableComponentModelTest() {
-    TableComponent model = context.resourceResolver().getResource(PATH + "/default")
-        .adaptTo(TableComponent.class);
+    // given
+    importTestTableResource();
+
+    // when
+    TableComponent model = toTableModel(PATH + "/default");
+
+    // then
     assertThat(model).isNotNull();
     assertThat(model.getTableClasses()).isEmpty();
     assertThat(model.isScrollable()).isFalse();
   }
 
   @Test
   void tableComponentTest() {
-    TableComponent model = context.resourceResolver().getResource(PATH + "/complex")
-        .adaptTo(TableComponent.class);
+    // given
+    importTestTableResource();
+
+    // when
+    TableComponent model = toTableModel(PATH + "/complex");
+
+    // then
     assertThat(model.isScrollable()).isTrue();
     assertThat(model.getTableClasses()).containsExactlyInAnyOrder("is-bordered", "is-striped",
         "is-narrow", "is-hoverable", "is-fullwidth", "is-scrollable");
   }
+
+  @Test
+  void shouldCalculateInitialColspansForTableTemplate() throws IOException {
+    // given
+    String jcrPath = "/content/table/template";
+    importMainTableTemplateResource(jcrPath);
```

Comment:
Testing against actual main template of the table (`src/main/resources/libs/kyanite/common/components/table/template/.content.json`).

This test wants to make sure, that this 3x3 table will receive colspan=4 by default for all its cells